src/test/java/com/marginallyclever/makelangelo/MissingTranslationTest.java

```diff
@@ -0,0 +1,129 @@
+package com.marginallyclever.makelangelo;
+
+import com.marginallyclever.util.PreferencesHelper;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MissingTranslationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(MissingTranslationTest.class);
+
+    private Pattern patternComment = Pattern.compile("^\\s*//.*");
+    private Pattern patternTranslator = Pattern.compile("Translator\\s*\\.\\s*get\\s*\\(\"(?<key>[^)]*)\"\\)");
+
+    @BeforeAll
+    public static void init() {
+        PreferencesHelper.start();
+        Translator.start();
+    }
+
+    @Test
+    public void findMissingTranslations() throws IOException {
+        List<String> results = new ArrayList<>();
+        File srcDir = new File("src" + File.separator + "main" + File.separator + "java");
+
+        List<File> files = listFiles(srcDir.toPath(), ".java");
+        // search in the files ...
+        files.forEach(file -> {
+            try {
+                searchInAFile(file, results);
+            } catch (IOException e) {
+                logger.warn("Can read file {}", file, e);
+            }
+        });
+
+        if (results.size() != 0) {
+            logger.info("translations missing:");
+            for (String result: results) {
+                logger.info("  {}", result);
+            }
+        }
+        assertEquals(0, results.size(), "Some translations are missing, see previous logs for details");
+    }
+
+    @Test
+    public void verifyThatMatcherWorks_results() throws IOException {
+        List<String> results = new ArrayList<>();
+
+        // matches.txt contains few entries with translation and one without. The list should not be empty
+        searchInAFile(new File("src/test/resources/translator/matches.txt"), results);
+
+        assertEquals(List.of("src/test/resources/translator/matches.txt:4: unknownKey"), results);
```

Comment:
If you run it on a windows systeme ( file char separator "\\" vs "/" ) ... (this is one of the current CI test failed for this PR : 
~~~
Error:    MissingTranslationTest.verifyThatMatcherWorks_results:68 expected: <[src/test/resources/translator/matches.txt:4: unknownKey]> but was: <[src\test\resources\translator\matches.txt:4: unknownKey]>
~~~
)
maybe juste 
`"src/test/resources/translator/matches.txt:4: unknownKey".replaceAll("/", File.separator)` 
?
or else convert to URI the File for the output on the file ?

Reply:
thx, I modified with `.replaceAll("/", File.separator)`